src/Input/Input.js

```diff
@@ -6,11 +6,44 @@ import classNames from 'classnames';
 import withStyles from '../styles/withStyles';
 import Textarea from './Textarea';
 
+export const isString = (value: ?(string | number)) => typeof value === 'string';
+export const isNumber = (value: ?(string | number)) => !isNaN(parseFloat(value));
+
+/**
+ * @param value
+ * @returns {boolean} true if string or number (including zero)
+ */
+export function hasValue(value: ?(number | string)) {
+  if (isString(value)) {
+    return true;
+  } else if (isNumber(value)) {
+    return true;
+  }
+  return !!value;
+}
+
+/**
+ * Determine if field is dirty (a.k.a. filled).
```

Comment:
The master branch was looking for invalid values instead of valid ones. Have you considered this option https://github.com/callemall/material-ui/blob/master/src/TextField/TextField.js#L110 ? Maybe we could make the implmentation shorter.